src/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeThread.Windows.cs

```diff
@@ -36,6 +42,208 @@ internal IntPtr[] GetWaitedHandleArray(int requiredCapacity)
             return _waitedHandles.Items;
         }
 
+        private void PlatformSpecificInitializeExistingThread()
+        {
+            _osHandle = GetOSHandleForCurrentThread();
+            _priority = MapFromOSPriority((OSThreadPriority)Interop.mincore.GetThreadPriority(_osHandle));
+        }
+
+        private static SafeWaitHandle GetOSHandleForCurrentThread()
+        {
+            SafeWaitHandle threadHandle;
+            if (Interop.mincore.GetCurrentThreadHandle(out threadHandle))
```

Comment:
It would be better to inline GetCurrentThreadHandle here so that it is obvious where is the last error coming from.